main.py

```
@@ -53,219 +87,1126 @@
+        return ReportResponse(
+            submission_id=submission_id,
+            status="submitted",
+            report_id=report.report_id,
+            timestamp=timestamp
+        )
+        
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.error(f"Report submission error: {e}")
+        raise HTTPException(status_code=500, detail=f"Report submission failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)